core/src/main/scala/scalaz/IndexedContsT.scala

```diff
@@ -145,10 +167,9 @@ abstract class IndexedContsTInstances extends IndexedContsTInstances0 {
       implicit val M: Functor[M] = M0
     }
 
-  implicit def ContsTMonadPlus[W[_], M[_], R](implicit W0: Comonad[W], M0: PlusEmpty[M]): MonadPlus[ContsT[W, M, R, ?]] =
-    new ContsTMonadPlus[W, M, R] {
+  implicit def FreeContsTMonad[W[_], S[_], R](implicit W0: Comonad[W]): Monad[ContsT[W, Free[S, ?], R, ?]] with BindRec[ContsT[W, Free[S, ?], R, ?]] =
+    new FreeContsTBindRec[W, S, R] with ContsTMonad[W, Free[S, ?], R] {
```

Comment:
`ContsTMonad` does not override `bind`. So it is not a problem.

Anyway, I swapped the order of super traits, ensuring that the final instance always pick stack-safe `bind` even if `ContsTMonad` provide another `bind` implementation.